dev_tools/nyaml2nxdl/nyaml2nxdl.py

```diff
@@ -0,0 +1,253 @@
+        if append.strip() == s.replace(".nxdl.xml", "")
+    ], "Your base class extension does not match any existing NeXus base classes"
+    tree = ET.parse(
+        os.path.join(nexus_def_path + "/base_classes", append + ".nxdl.xml")
+    )
+    root = tree.getroot()
+    # warning: tmp files are printed on disk and removed at the ends!!
+    pretty_print_xml(root, "tmp.nxdl.xml")
+    input_tmp_xml = "tmp.nxdl.xml"
+    out_tmp_yml = "tmp_parsed.yaml"
+    converter = Nxdl2yaml([], [])
+    converter.print_yml(input_tmp_xml, out_tmp_yml, verbose)
+    nyaml2nxdl(input_file=out_tmp_yml, out_file="tmp_parsed.nxdl.xml", verbose=verbose)
+    tree = ET.parse("tmp_parsed.nxdl.xml")
+    tree2 = ET.parse(input_file)
+    root_no_duplicates = ET.Element(
+        "definition",
+        {
+            "xmlns": "http://definition.nexusformat.org/nxdl/3.1",
+            "xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
+            "xsi:schemaLocation": "http://www.w3.org/2001/XMLSchema-instance",
+        },
+    )
+    for attribute_keys in root.attrib.keys():
+        if (
+            attribute_keys
+            != "{http://www.w3.org/2001/XMLSchema-instance}schemaLocation"
+        ):
+            attribute_value = root.attrib[attribute_keys]
+            root_no_duplicates.set(attribute_keys, attribute_value)
+    for elems in root.iter():
+        if "doc" in elems.tag:
+            root_doc = ET.SubElement(root_no_duplicates, "doc")
+            root_doc.text = elems.text
+            break
+    group = "{http://definition.nexusformat.org/nxdl/3.1}group"
+    root_no_duplicates = compare_niac_and_my(
+        tree, tree2, verbose, group, root_no_duplicates
+    )
+    field = "{http://definition.nexusformat.org/nxdl/3.1}field"
+    root_no_duplicates = compare_niac_and_my(
+        tree, tree2, verbose, field, root_no_duplicates
+    )
+    attribute = "{http://definition.nexusformat.org/nxdl/3.1}attribute"
+    root_no_duplicates = compare_niac_and_my(
+        tree, tree2, verbose, attribute, root_no_duplicates
+    )
+    pretty_print_xml(
+        root_no_duplicates,
+        f"{input_file.replace('.nxdl.xml', '')}" f"_appended.nxdl.xml",
+    )
+
+    input_file_xml = input_file.replace(".nxdl.xml", "_appended.nxdl.xml")
+    out_file_yml = input_file.replace(".nxdl.xml", "_appended_parsed.yaml")
+    converter = Nxdl2yaml([], [])
+    converter.print_yml(input_file_xml, out_file_yml, verbose)
+    nyaml2nxdl(
+        input_file=out_file_yml,
+        out_file=out_file_yml.replace(".yaml", ".nxdl.xml"),
+        verbose=verbose,
+    )
+    os.rename(
+        f"{input_file.replace('.nxdl.xml', '_appended_parsed.yaml')}",
+        f"{input_file.replace('.nxdl.xml', '_appended.yaml')}",
+    )
+    os.rename(
+        f"{input_file.replace('.nxdl.xml', '_appended_parsed.nxdl.xml')}",
+        f"{input_file.replace('.nxdl.xml', '_appended.nxdl.xml')}",
+    )
+    os.remove("tmp.nxdl.xml")
```

Comment:
As pathlib objects, these lines would be of the form `PATHLIB_OBJECT.unlink()`

Reply:
Code removed